### PR TITLE
🍓 refactor

### PR DESCRIPTION
**Pull request by ZackAmes**

### CODE REVIEW

**Comment by 0xMugen** [comment]
We fixed the movement slippage in the previous pr #41 

The error was using floor rather than truncate in this file:
```typescript
  export function truncateToDecimals(num: number, decimalPlaces: number) {
    const multiplier = Math.pow(10, decimalPlaces)
    return Math.trunc(num * multiplier) / multiplier
  }
```

Can you list the changes you've done, specificaly for the contract side if it affects the call data or any logic in the front end and if so the needed changes in the front end

File: refactor/src/lib/ui/Ui.svelte
Comment (0xMugen) [inline refactor/src/lib/ui/Ui.svelte:1]: Good QOL stuff

File: refactor/src/lib/helper.ts
Comment (0xMugen) [inline refactor/src/lib/helper.ts:1]: Refactor

File: refactor/src/lib/3d/utils/moveUtils.ts
Comment (0xMugen) [inline refactor/src/lib/3d/utils/moveUtils.ts:1]: Handle The rebase. TO not conflict with the 1on1 changes

File: contracts/src/models/bullet.cairo
Comment (0xMugen) [inline contracts/src/models/bullet.cairo:1]: good

**Review by 0xMugen — APPROVED** [review]
Just handle the rebase correctly. And we are good

**Comment by ZackAmes** [comment]
Includes cleaning up the move utils with constants, store the submoves on the 100,000x100,000 scale and then only scaling down when rendering, ui improvements such as entering first person when you start recording and only showing the move ui when you are turn player

**Pull request merged**